gc/base/standard/Scavenger.cpp

```diff
@@ -1756,7 +1756,7 @@ MM_Scavenger::incrementalScavengeObjectSlots(MM_EnvironmentStandard *env, omrobj
 
 	/* Set up for maintaining hot field stats for scalar objects */
 	MM_ScavengerHotFieldStats *hotFieldStats = NULL;
-	if (!objectScanner->isIndexableObject() && _extensions->scavengerTraceHotFields) {
+	if ((!objectScanner->isIndexableObject()) && (NULL != env->_hotFieldStats)) {
```

Comment:
extra pair of brackets around !objectScanner->isIndexableObject() is not necessary as far as it still be bool:
if (!objectScanner->isIndexableObject() && (NULL != env->_hotFieldStats)) {

Reply:
👍